src/t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_conn_vertex_to_tree.hxx

```diff
@@ -0,0 +1,245 @@
+class t8_cmesh_vertex_conn_tree_to_vertex;
+
+/** This class stores the vertex to tree lookup for
+ * global vertex indices for a cmesh.
+ * Thus, given a global vertex id the class provides
+ * information about the trees the vertex belongs to and 
+ * the corresponding local vertex ids inside these trees.
+ * 
+ * In particular, this class stores the lookup
+ *
+ * global_vertex_id -> List of (tree, tree_local_vertex)
+ *
+ * for a cmesh.
+ * It is the opposite lookup as \ref t8_cmesh_vertex_conn_tree_to_vertex.hxx
+ *
+ * The global vertex ids must not be contiguous, that is, we have some set
+ *
+ * {I_0 < I_1 < ...< I_N} of natural numbers corresponding to the N+1 vertices.
+ *
+ * I_0 does not have to be 0 and I_N does not have to be N.
+ *
+ * So we need lookup: I_i -> i
+ *  store this in a hash table.
+ *
+ * datatypes:
+ *
+ * global id: t8_gloidx_t
+ * (tree_id, tree_vertex): std::pair<t8_locidx_t, int> = TV_PAIR
+ * List of (tree_id, tree_vertex): std::vector<PAIR> = TV_LIST
+ * Table global_id -> TV_LIST: std::unordered_map<t8_gloidx_t, TV_LIST>
+ *
+ */
+class t8_cmesh_vertex_conn_vertex_to_tree {
+ public:
+  /** Standard constructor.
+   * Initializes the class and allows setting vertex entries
+   * via \ref add_vertex_to_tree
+   */
+  t8_cmesh_vertex_conn_vertex_to_tree (): state (INITIALIZED)
+  {
+  }
+
+  /** Function to fill vtt from cmesh and ttv information.
+   * Sets all global ids and associated tree vertices from
+   * the given input class.
+   * Afterwards, the class is set to committed and can be used.
+   *
+   * \param [in] cmesh A committed cmesh with set tree to vertex entries.
+   * \param [in] ttv A filled tree to vertex list for \a cmesh.
+  */
+  void
+  build_from_ttv (const t8_cmesh_t cmesh, t8_cmesh_vertex_conn_tree_to_vertex& ttv);
+
+  /** Variable type for (tree_id, tree_vertex_id) pair */
+  using tree_vertex_pair = std::pair<t8_locidx_t, int>;
+
+  /** list of tree vertex pairs, each global vertex id maps to
+    * such a list. */
+  using tree_vertex_list = std::vector<tree_vertex_pair>;
+
+  /** The internal storage data type used for storing the vertex to tree data. */
+  using vtt_storage_type = std::unordered_map<t8_gloidx_t, tree_vertex_list>;
+
+  /* Setter functions */
+  /** Given a cmesh, build up the vertex_to_tree.
+   * \param [in] cmesh An initialized but not yet committed cmesh.
+   * \return: some error value to be specified.
+   * The cmesh must not be committed, but all tree information and neighbor information must
+   * have been set.
+   * Currently, \a cmesh has to be replicated. */
+  void
+  set_vertex_to_tree_list (const t8_cmesh_t cmesh);
+
+  /** Get the list of global trees and local vertex ids a global vertex is connected to.
+   * 
+   * \param [in] global_vertex_id The global id of a vertex in the cmesh.
+   * \return The list of global tree ids and local vertex ids of \a global_vertex_id.
```

Comment:
There is a documentation mismatch.
This says "list of global tree ids" but the return value tree_vertex_list stores local ids.
@holke double check this.

Reply:
double checked it. The comment is wrong, we are storing local tree ids.
Corrected the comment.